examples/spring-boot-webflux/src/main/java/example/springframework/boot/webflux/HelloConfiguration.java

```diff
@@ -0,0 +1,71 @@
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerHttpClientBuilder;
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerStrategy;
+import com.linecorp.armeria.client.retry.RetryStrategy;
+import com.linecorp.armeria.client.retry.RetryingHttpClient;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.docs.DocService;
+import com.linecorp.armeria.server.logging.AccessLogWriter;
+import com.linecorp.armeria.server.logging.LoggingService;
+import com.linecorp.armeria.spring.ArmeriaServerConfigurator;
+import com.linecorp.armeria.spring.web.reactive.ArmeriaClientConfigurator;
+
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+
+/**
+ * An example of a configuration which provides beans for customizing the server and client.
+ */
+@Configuration
+public class HelloConfiguration {
+
+    /**
+     * A user can configure a {@link Server} by providing an {@link ArmeriaServerConfigurator} bean.
+     */
+    @Bean
+    public ArmeriaServerConfigurator armeriaServerConfigurator() {
+        // Customize the server using the given ServerBuilder. For example:
+        return builder -> {
+            // Add DocService that enables you to send Thrift and gRPC requests from web browser.
+            builder.serviceUnder("/docs", new DocService());
+
+            // Log every message which the server receives and responds.
+            builder.decorator(LoggingService.newDecorator());
+
+            // Write access log after completing a request.
+            builder.accessLogWriter(AccessLogWriter.combined(), false);
+
+            // You can also bind asynchronous RPC services such as Thrift and gRPC:
+            // builder.service(THttpService.of(...));
+            // builder.service(new GrpcServiceBuilder()...build());
+        };
+    }
+
+    /**
+     * A user can configure a {@link Client} by providing an {@link ArmeriaClientConfigurator} bean.
+     */
+    @Bean
+    public ArmeriaClientConfigurator armeriaClientConfigurator() {
+        // Customize the client using the given HttpClientBuilder. For example:
+        return builder -> {
+            // Use a circuit breaker for each remote host.
+            final CircuitBreakerStrategy strategy = CircuitBreakerStrategy.onServerErrorStatus();
+            builder.decorator(new CircuitBreakerHttpClientBuilder(strategy).newDecorator());
+
+            // Automatically retry a request when the server returns a 5xx response.
+            builder.decorator(RetryingHttpClient.newDecorator(RetryStrategy.onServerErrorStatus()));
+
+            // Use a custom client factory in order to disable the certificate validation,
+            // which means any certificate received from the server will be accepted.
+            final ClientFactory clientFactory = new ClientFactoryBuilder().sslContextCustomizer(
+                    b -> b.trustManager(InsecureTrustManagerFactory.INSTANCE)).build();
```

Comment:
This client factory will not be cleaned up when application context is terminated. How about providing it as a bean so that Spring calls `ClientFactory.close()` for us when the application terminates?